.github/workflows/tests.yml

```diff
@@ -24,16 +32,14 @@ jobs:
 
   gen-test-matrix:
     name: Find test files
-    if: "!contains(github.event.head_commit.message, '[skip ci]')"
-    runs-on: [self-hosted]
+    runs-on: ubuntu-latest
     outputs:
       matrix: ${{ steps.set-matrix.outputs.matrix }}
-    strategy:
-      fail-fast: false
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
```

Comment:
Only tangentially related to this PR, without having to do this manually, you can get [Dependabot](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file) to automatically open PRs to update the workflows, see for example https://github.com/JuliaPhysics/Measurements.jl/blob/21cf93c7dc25b7a1b14d5b877bc8f4884599ff5e/.github/dependabot.yml